internal/core/domain/claim.go

```diff
@@ -241,3 +241,12 @@ func (c *Claim) GetCredentialStatus() (*verifiable.CredentialStatus, error) {
 	}
 	return cStatus, nil
 }
+
+// GetPublicKey returns the public key of the claim
+// If the schema is not supported, it returns an unSupportedPublicKeyType
+func (c *Claim) GetPublicKey() PublicKey {
```

Comment:
@martinsaporiti I still insist to move this function to utilities, as you should have too much context to call it properly for one specific claim, this is knowledge that doesn't exist on the Claim level